### PR TITLE
fix: paginate GitHub API responses to fetch all items

### PR DESCRIPTION
**Pull request by ushironoko**
## Summary

- GitHub REST APIのデフォルト30件制限を解消するため、`gh_api_paginate()`関数を追加（`--paginate --slurp --jq 'add // []'`で全ページ取得・マージ）
- `fetch_changed_files`, `fetch_review_comments`, `fetch_discussion_comments`, `fetch_reviews`をページネーション対応に変更
- 各エンドポイントに`per_page=100`を付与し、APIコール数を最小化

## Test plan

- [x] `cargo test` 全テスト通過（315 passed）
- [x] 30ファイル以上のPRで全ファイルが表示されることを確認
- [x] 30件以上のレビューコメントがあるPRで全コメントが取得されることを確認

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by ushironoko — COMMENTED** [review]
[AI Rally - Reviewer]

Pagination is wired correctly for the targeted GitHub list endpoints, and local verification is green (`cargo test` passed: 319 passed, 1 ignored). I found no blocking issues for merge; only minor robustness/test coverage suggestions.

File: src/github/client.rs [hunk 1]
```diff
@@ -93,6 +93,22 @@ pub async fn gh_api(endpoint: &str) -> Result<serde_json::Value> {
...
 
+/// Execute gh api command with automatic pagination for array endpoints.
+/// Fetches all pages and merges into a single JSON array.
+/// Caller should include `per_page=100` in endpoint if desired.
+pub async fn gh_api_paginate(endpoint: &str) -> Result<serde_json::Value> {
+    let output = gh_command(&["api", "--paginate", "--slurp", endpoint]).await?;
+    let pages: Vec<serde_json::Value> =
+        serde_json::from_str(&output).context("Failed to parse gh api paginated response")?;
+    let mut result = Vec::new();
+    for page in pages {
```
Comment (ushironoko) [inline src/github/client.rs]: [AI Rally - Reviewer]

`gh_api_paginate` silently ignores non-array pages. Consider failing fast when a page is not `Value::Array` so unexpected response-shape changes (or future misuse on non-array endpoints) do not become silent data loss.

File: src/github/client.rs [hunk 2]
```diff
@@ -93,6 +93,22 @@ pub async fn gh_api(endpoint: &str) -> Result<serde_json::Value> {
     serde_json::from_str(&output).context("Failed to parse gh api response as JSON")
 }
 
+/// Execute gh api command with automatic pagination for array endpoints.
+/// Fetches all pages and merges into a single JSON array.
+/// Caller should include `per_page=100` in endpoint if desired.
+pub async fn gh_api_paginate(endpoint: &str) -> Result<serde_json::Value> {
```
Comment (ushironoko) [inline src/github/client.rs]: [AI Rally - Reviewer]

Please add focused unit tests for `gh_api_paginate` (multi-page flattening, empty result, and unexpected page shape). This helper is now critical for PR data completeness and currently has no direct test coverage.
